substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jfr/JfrTypeRepository.java

```diff
@@ -246,10 +251,14 @@ private boolean isClassVisited(TypeInfo typeInfo, Class<?> clazz) {
         return typeInfo.classes.contains(clazz) || flushedClasses.contains(clazz);
     }
 
-    private boolean addPackage(TypeInfo typeInfo, Package pkg, Module module) {
+    private boolean addPackage(TypeInfo typeInfo, Package pkg, Module module, boolean generated) {
         if (isPackageVisited(typeInfo, pkg)) {
-            assert module == (flushedPackages.containsKey(pkg.getName()) ? flushedPackages.get(pkg.getName()).module : typeInfo.packages.get(pkg.getName()).module);
-            return false;
+            Module cached = (flushedPackages.containsKey(pkg.getName()) ? flushedPackages.get(pkg.getName()).module : typeInfo.packages.get(pkg.getName()).module);
```

Comment:
The logic here was a bit confusing  and has been removed now in favor of what's described in the above response.

Previously, this logic was so that we would only replace the cached module if it is the unnamed module. And we only want to bother replacing it with a named module. This effectively would result in only serializing named modules, or the unnamed module if there was no named module for the given package.